crates/datafusion/src/utils/exprs_to_filter.rs

```diff
@@ -0,0 +1,346 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use arrow_array::{Array, Scalar};
+use arrow_schema::SchemaRef;
+use datafusion::logical_expr::Operator;
+use datafusion_expr::{BinaryExpr, Expr};
+use hudi_core::exprs::{HudiOperator, PartitionFilter};
+use std::sync::Arc;
+
+// TODO: Handle other Datafusion `Expr`
+
+/// Converts a slice of DataFusion expressions (`Expr`) into a vector of `PartitionFilter`.
+/// Returns `Some(Vec<PartitionFilter>)` if at least one filter is successfully converted,
+/// otherwise returns `None`.
+pub fn convert_exprs_to_filter(
+    filters: &[Expr],
+    partition_schema: &SchemaRef,
+) -> Vec<PartitionFilter> {
+    let mut partition_filters = Vec::new();
+
+    for expr in filters {
+        match expr {
+            Expr::BinaryExpr(binary_expr) => {
+                if let Some(partition_filter) = convert_binary_expr(binary_expr, partition_schema) {
+                    partition_filters.push(partition_filter);
+                }
+            }
+            Expr::Not(not_expr) => {
+                // Handle NOT expressions
+                if let Some(partition_filter) = convert_not_expr(not_expr, partition_schema) {
+                    partition_filters.push(partition_filter);
+                }
+            }
+            _ => {
+                continue;
+            }
+        }
+    }
+
+    partition_filters
+}
+
+/// Converts a binary expression (`Expr::BinaryExpr`) into a `PartitionFilter`.
+fn convert_binary_expr(
```

Comment:
instead of helper functions, can we implement `TryFrom` for Hudi's generic filter struct to convert Expr to Hudi filter?